### ISSUE TITLE
"Failed to read seal configuration" errors after upgrading to 0.10.2

### ISSUE THREAD

**Issue by mjacksonw** [hashicorp/vault#4721]
**Describe the bug**
After upgrading from 0.10.1 (or 0.10.0) to 0.10.2, vault produces the following errors:

```
[ERROR] core: failed to read seal configuration: error="UnmarshalTypeError: cannot unmarshal string into Go value of type []uint8"
[INFO ] core: security barrier not initialized 
```

**Environment:**
* Vault Server Version (retrieve with `vault status`): 0.10.2
* Vault CLI Version (retrieve with `vault version`): 0.10.2
* Server Operating System/Architecture: Alpine Linux 3.6 / Docker

Vault server configuration file(s):

```hcl
listener "tcp" {
  address     = "0.0.0.0:8200"
  tls_disable = "true"
  proxy_protocol_behavior = "use_always"
  proxy_protocol_authorized_addrs = "REDACTED"
}

storage "dynamodb" {
  ha_enabled      = "true"
  region          = "us-east-1"
  table           = "REDACTED"
  advertise_addr  = "REDACTED"
  recovery_mode   = 1
}

disable_mlock = "true"
ui = "true"
```

**Comment by jefferai** [comment]
Can you build a branch if I put in some debug?

**Comment by mjacksonw** [comment]
I can give it a try! Also, I just spun up a local version of 0.10.2 (on MacOS) and got the same error, so it's not likely related to the environment.

**Comment by mjacksonw** [comment]
Yup, got a build env going, and confirmed I get the same issue against master (which is likely unsurprising).

**Comment by mjacksonw** [comment]
Did a `git bisect` against the diff between 0.10.1 and 0.10.2 and came up against this seemingly-relevant commit:

https://github.com/hashicorp/vault/commit/39f2d66eb7fb7cca3438aa8602993bc42f4dd475

**Comment by kalafut** [comment]
I landed at that same commit too through other means. I ran a forked version of our tests using the old write, but the new read, and get the error you reported. We're discussing next steps now.

**Comment by jefferai** [comment]
This will be fixed in 0.10.3.

**Comment by kalafut** [comment]
@mjacksonw Now that this is merged, if you get a chance to retest with master it would be a very useful confirmation that this fully fixed your issue.
